src/main/java/com/hpe/application/automation/tools/results/RichReportAction.java

```diff
@@ -0,0 +1,158 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * ___________________________________________________________________
+ *
+ */
+
+package com.hpe.application.automation.tools.results;
+
+import hudson.FilePath;
+import hudson.model.Action;
+import hudson.model.DirectoryBrowserSupport;
+import hudson.model.Run;
+import hudson.tasks.test.TestResultProjectAction;
+import jenkins.tasks.SimpleBuildStep;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class RichReportAction implements Action, SimpleBuildStep.LastBuildAction {
+
+    private static final String RICH_REPORT_FOLDER = "RichReport";
+    private static final String REPORT_INDEX = "report.index";
+
+    private Map<String, DetailReport> richReportMap = new LinkedHashMap<String, DetailReport>();
+    private final List<TestResultProjectAction> projectActionList;
+
+
+    private final Run<?,?> build;
+
+    public RichReportAction(Run<?,?> build) throws IOException {
+        this.build = build;
+        File reportFolder = new File(build.getRootDir(), RICH_REPORT_FOLDER);
+        if (reportFolder.exists()) {
+            File indexFile = new File(reportFolder, REPORT_INDEX);
+            if (indexFile.exists()) {
+                File file = new File(build.getRootDir(), RICH_REPORT_FOLDER);
+                DirectoryBrowserSupport dbs = new DirectoryBrowserSupport(this, new FilePath(file), "report", "graph.gif", false);
+
+
+                createRichIndexFile(build, indexFile, dbs);
+            }
+        }
+        projectActionList = new ArrayList<TestResultProjectAction>();
+    }
+
+    private void createRichIndexFile(Run<?, ?> build, File indexFile, DirectoryBrowserSupport dbs) throws IOException {
+        BufferedReader br = new BufferedReader(new FileReader(indexFile));
+        String line;
+        boolean rolling = true;
+        while ((line = br.readLine()) != null) {
+            String[] values = line.split("\t");
+            if (values.length < 1)
+                continue;
+            DetailReport report = new DetailReport(build, values[0], dbs);
+            if (rolling) {
```

Comment:
Those if/else statements are really hard to understand, maybe wrap them with a function describing what is their purpose?